src/plugins/componentEventListenersPlugin.ts

```diff
@@ -0,0 +1,41 @@
+import { Plugin, PluginKey } from "prosemirror-state";
```

Comment:
If you want to make this diff smaller you could do this reorganization first, even if this will be the only plugin for a moment.